lib/iris/fileformats/netcdf.py

```diff
@@ -1327,7 +1328,10 @@ def _cf_coord_identity(self, coord):
             elif coord.standard_name == "longitude":
                 units = 'degrees_east'
 
-        return (coord.standard_name, coord.long_name, units)
+        elif isinstance(coord.coord_system, iris.coord_systems.RotatedGeogCS):
+            units = 'degrees'
```

Comment:
@lbdreyer As per the CF spec sections 4.1 and 4.2.

@djkirkham and @marqh This reverts your changes from #1943. Can we all agree this is what we want for rotated pole units, as per CF.